cmd/kube-controller-manager/app/core.go

```diff
@@ -143,14 +152,33 @@ func startNodeIpamController(ctx ControllerContext) (http.Handler, bool, error)
 		}
 	}
 
+	var nodeCIDRMaskSizeIPv4, nodeCIDRMaskSizeIPv6 int
+	if utilfeature.DefaultFeatureGate.Enabled(kubefeatures.IPv6DualStack) {
+		setNodeCIDRMaskSizes(ctx.ComponentConfig.NodeIPAMController, &nodeCIDRMaskSizeIPv4, &nodeCIDRMaskSizeIPv6)
+	} else {
+		setNodeCIDRMaskSizesDualStack(ctx.ComponentConfig.NodeIPAMController, &nodeCIDRMaskSizeIPv4, &nodeCIDRMaskSizeIPv6)
+	}
+
+	// IPv4 node cidr mask size not defined, set to defaultNodeMaskCIDRIPv4
```

Comment:
this is not accurate because we are only checking if the mask is defined if the dualStack feature  gate is enabled per L156

Reply:
`setNodeCIDRMaskSizes` or `setNodeCIDRMaskSizesDualStack` will set the mask size based on user input. If the user input for mask sizes isn't provided, then here we set the values to default.

https://github.com/kubernetes/kubernetes/pull/79993/files#diff-0ca8b7870d367d5fc309a2e737a6eed2R563-R575
https://github.com/kubernetes/kubernetes/pull/79993/files#diff-0ca8b7870d367d5fc309a2e737a6eed2R577-R591

Reply:
you can resolve this @aramase , it was related to the order the previous if was checking the mask, my fault it was really bad worded :sweat_smile:

Reply:
sounds good!